src/main/java/org/opensearch/knn/index/KNNSettings.java

```diff
@@ -66,6 +66,7 @@ public class KNNSettings {
      * Settings name
      */
     public static final String KNN_SPACE_TYPE = "index.knn.space_type";
+    public static final String INDEX_KNN_BUILD_VECTOR_DATA_STRUCTURE_THRESHOLD = "index.knn.build_vector_data_structure_threshold";
```

Comment:
not a blocker, but I think we need to rename this setting. Need to brainstorm some ideas, but this is too verbose.

Some thoughts
1. index.knn.exact.threshold
2. index.ann.threshold

Reply:
@VijayanB lets take this as AI to conclude on the naming.

Reply:
Sure.